draft-barnes-sframe-mls.md

```diff
@@ -57,17 +87,25 @@ sender_base_key[index] = HKDF-Expand(sframe_epoch_secret,
                            encode_big_endian(index, 4), AEAD.Nk)
 ~~~~~
 
-For compactness, do not send the whole epoch number.  Instead, we send only its
-low-order E bits.  Note that E effectively defines a re-ordering window, since
-no more than 2^E epoch can be active at a given time.  Receivers MUST be
-prepared for the epoch counter to roll over, removing an old epoch when a new
-epoch with the same E lower bits is introduced.  (Sender indices cannot be
-similarly compressed.)
+The Key ID (KID) field in the SFrame header provides the epoch and index values
+that are needed to generate the appropriate key from the MLS key schedule.
 
 ~~~~~
 KID = (sender_index << E) + (epoch % (1 << E))
 ~~~~~
 
+For compactness, do not send the whole epoch number.  Instead, we send only its
+low-order E bits.  The participants in the group MUST agree on the value of E
+for a given session, through some negotiation not specified here.
+
+Note that E effectively defines a re-ordering window, since no more than 2^E
+epoch can be active at a given time.  The better the participants are in sync
+with regard to key roll-over, and the less reordering of SFrame-protected
+payloads by the network, the fewer bits of epoch are necessary.
+
+Receivers MUST be prepared for the epoch counter to roll over, removing an old
+epoch when a new epoch with the same E lower bits is introduced.
+
```

Comment:
New joiners might have problems when joining towards the end of a rollover window, because they might not know if they are in the current window or already in the next one. Limited trial decryption might be the solution here.

Reply:
I don't *think* this is an problem, but I'll add an open issue.